enhancements/subscription-content/subscription-injection.md

```diff
@@ -0,0 +1,662 @@
+  containers:
+  - name: main
+    ...
+    volumeMounts:
+    # Mount yum-repos.d configs in /run/secrets
+    - name: yum-repo
+      mountPath: /run/secrets
+    # Mount entitlements in /run/secrets/etc-pki-entitlement
+    - name: etc-pki-entitlement
+      mountPath: /run/secrets/etc-pki-entitlement
+  ...
+  volumes:
+  # yumRepositories are consolidated into a single volume
+  # Projected volumes with no key/path mappings require that all refereneced keys are unique
+  - name: yum-repo
+    projected:
+      sources:
+      - configMap:
+          name: local-repo
+      - configMap:
+          name: driver-repo
+  # Entitlement keys are likewise consolidated into a single volume
+  # Projected volumes with no key/path mappings require that all referenced keys are unique
+  - name: etc-pki-entitlement
+    projected:
+      sources:
+      - secret:
+          name: local-entitlement
+      - secret:
+          name: driver-entitlement
+```
+
+Injecting a cluster-scoped `ClusterBundle`:
+
+```yaml
+kind: Pod
+apiVersion: core/v1
+metadata:
+  annotations:
+    # Inject cluster-wide bundles
+    "subscription.openshift.io/inject-cluster-bundle": cluster,extra
+    # Record the generation of the injected bundles
+    "subscription.openshift.io/bundle-generations": |
+      {
+        "clusterBundles": {
+          "cluster": 1,
+          "extra": 4
+        }
+      }
+spec:
+  containers:
+  - name: main
+    ...
+    volumeMounts:
+    # Mount yum-repos.d configs in /run/secrets
+    - name: yum-repo
+      mountPath: /run/secrets
+    # Mount entitlements in /run/secrets/etc-pki-entitlement
+    - name: etc-pki-entitlement
+      mountPath: /run/secrets/etc-pki-entitlement
+  ...
+  volumes:
+  # yumRepositories are consolidated into a single volume
+  # Projected volumes with no key/path mappings require that all refereneced keys are unique
+  - name: yum-repo
+    csi:
+      driver: projected-resource.storage.openshift.io
+      volumeAttributes:
+        sources:
+        - share:
+            name: cluster-repo
+        - share:
+            name: extra-repo
+  # Entitlement keys are likewise consolidated into a single volume
+  # Projected volumes with no key/path mappings require that all referenced keys are unique
+  - name: etc-pki-entitlement
+    csi:
+      driver: projected-resource.storage.openshift.io
+      volumeAttributes:
+        sources:
+        - share:
+            name: local-entitlement
+        - share:
+            name: driver-entitlement
+```
+
+Allow/deny lists for entitlements that are always injected:
+
+```yaml
+kind: Pod
+apiVersion: core/v1
+metadata:
+  annotations:
+    # Allow some bundles in the namespace, deny all others
+    subscription.openshift.io/allow-bundles: "bundle-A,bundle-B"
+    # Deny some bundles in the namespace, but allow others
+    subscription.openshift.io/deny-bundles: "bundle-B,bundle-C"
+    # Deny all bundles in the namespace
+    subscription.openshift.io/deny-bundles: "*"
```

Comment:
so to be clear the default behavior if you don't specify any allow/deny annotation is to allow `*`, right?

Reply:
Yes - I'll add a note.